pactflow/client.ts

```diff
@@ -61,16 +88,90 @@ export class PactflowClient implements Client {
       );
     }
 
-    async review(body: RefineInput): Promise<RefineResponse> {
-      // submit review request
+    /**
+     * Resolve the OpenAPI specification from the provided input.
+     * @param remoteOpenAPIDocument The remote OpenAPI document to resolve.
+     * @returns The resolved OpenAPI document.
+     * @throws Error if the resolution fails.
+     */
+    async resolveOpenAPISpec(remoteOpenAPIDocument: RemoteOpenAPIDocument): Promise<OpenAPI> {
+      const openAPISchema = RemoteOpenAPIDocumentSchema.safeParse(remoteOpenAPIDocument);
+      
+      if (openAPISchema.success && remoteOpenAPIDocument) {
+        const unresolvedSpec = await this.getRemoteSpecContents(openAPISchema.data);
+        const resolvedSpec = await Swagger.resolve({ spec: unresolvedSpec });
+
+        if (!resolvedSpec.errors?.length) {
+          return resolvedSpec.spec;
+        }
+
+        throw new Error(`Failed to resolve OpenAPI document: ${resolvedSpec.errors?.join(", ")}`);
+      }
+
+      throw new Error(`Invalid RemoteOpenAPIDocument: ${JSON.stringify(openAPISchema.error?.issues)}`);
+    }
+
+    /**
+     * Fetch the contents of a remote OpenAPI document.
+     * @param openAPISchema The schema for the remote OpenAPI document.
+     * @returns A promise that resolves to a map of the OpenAPI document contents.
+     * @throws Error if the URL is not provided or the fetch fails.
+     */
+    async getRemoteSpecContents(openAPISchema: RemoteOpenAPIDocument): Promise<any> {
+      if (openAPISchema.url) {
+        let headers = {};
+        if (openAPISchema.authToken) {
+          headers = {
+            Authorization: `${openAPISchema.authScheme ?? "Bearer"} ${openAPISchema.authToken}`,
+          };
+        }
+
+        const remoteSpec = await fetch(openAPISchema.url, {
+          headers,
+          method: "GET",
+        });
+
+        switch(remoteSpec.headers.get("Content-Type")?.toLowerCase()) {
+          case "application/json":
+            return await remoteSpec.json();
+          case "text/plain":
+          case "text/plain;charset=utf-8":
+          case "text/plain; charset=utf-8":
+          case "application/yaml":
+            return JSON.parse(JSON.stringify(yaml.load(await remoteSpec.text()), null, 2));
+          default:
+            throw new Error(`Unsupported Content-Type: ${remoteSpec.headers.get("Content-Type")}`);
+        }
```

Comment:
I would not match the content type this way as you'll be bound to miss something. For example:

- Other charsets: `application/json; charset=ascii`, `application/json;charset=utf-16`, etc.
- Custom content types: `application/vnd.custom.openapi+json`, `application/vnd.custom.openapi+yaml`, or maybe even `application/openapi+json`

Instead:

1. I would trust that `charset` to be used to parse the input to text correctly by the underlying library
2. I would try to deserialise as JSON first, and automatically fallback to YAML irrespective of the content type. 
3. If both deserialisations fail, then throw an error.